flowmachine/tests/test_label_event_score.py

```diff
@@ -0,0 +1,233 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from flowmachine.core import JoinToLocation
+from flowmachine.features import EventScore
+from flowmachine.features.subscriber.label_event_score import LabelEventScore
+
+
+@pytest.mark.usefixtures("skip_datecheck")
+def test_labelled_event_score_column_names(
+    exemplar_level_param, get_column_names_from_run
+):
+    if exemplar_level_param["level"] not in JoinToLocation.allowed_levels:
+        pytest.skip(f'{exemplar_level_param["level"]} not valid for this test')
+    es = EventScore(start="2016-01-01", stop="2016-01-05", **exemplar_level_param)
+    labelled = LabelEventScore(scores=es, required="evening")
+    assert get_column_names_from_run(labelled) == labelled.column_names
+
+
+def test_locations_are_labelled_correctly(get_dataframe):
+    """
+    Test whether locations are labelled corrected.
+    """
+    es = EventScore(start="2016-01-01", stop="2016-01-05", level="versioned-site")
+
+    ls = LabelEventScore(
+        scores=es,
+        labels={
+            "daytime": {
+                "type": "Polygon",
+                "coordinates": [[[-1.1, -1.1], [-1, 1.1], [1.1, 1.1], [1.1, -1.1]]],
+            }
+        },
+    )
+    df = get_dataframe(ls)
+    assert list(df["label"].unique()) == ["daytime"]
+
+
+def test_whether_passing_reserved_label_fails():
+    """
+    Test whether passing the reserved label 'unknown' fails.
+    """
+    es = EventScore(start="2016-01-01", stop="2016-01-05", level="versioned-site")
+
+    with pytest.raises(ValueError):
+        ls = LabelEventScore(
+            scores=es,
+            labels={
+                "unknown": {
+                    "type": "Polygon",
+                    "coordinates": [
+                        [
+                            [-1.1, -1.1],
+                            [-1.0, 1.1],
+                            [1.1, 1.1],
+                            [1.1, -1.1],
+                            [-1.1, -1.1],
+                        ]
+                    ],
+                }
+            },
+        )
+
+
+def test_whether_required_label_relabels(get_dataframe):
+    """
+    Test whether required label relabel the location of subscribers who did not originally have the required label.
```

Comment:
Hmm, I'm still not sure whether I understand what the meaning/purpose of the required label actually is. Does this mean it unconditionally relabels locations of subscribers to this? Or only previously unassigned locations?